store/store.go

```diff
@@ -561,6 +575,10 @@ func (s *Store) newRequest(reqOptions *requestOptions, user *auth.UserState) (*h
 	req.Header.Set("X-Ubuntu-Series", s.series)
 	req.Header.Set("X-Ubuntu-Wire-Protocol", UbuntuCoreWireProtocol)
 
+	if os.Getenv("SNAPPY_DELTA_DOWNLOAD_FORMATS") != "" {
```

Comment:
Actually, it feels like this shouldn't be an external setting at all. snapd should internally know which delta formats it recognizes, and provide that to the server when necessary. Why would it be a good idea to have an external environment variable telling what snapd ought to consume here?